### PR TITLE
check ptr against NULL before dereferencing it

### PR DESCRIPTION
**Pull request by kokke**
This is a proposed fix for https://github.com/curl/curl/issues/6709

There is a small issue of dereferencing a pointer before checking it against NULL.

From [tests/libtest/lib1536.c : 72 - 76](https://github.com/curl/curl/blob/master/tests/libtest/lib1536.c#L72)

```C
72  if(memcmp(scheme, "HTTP", 5) != 0) {
              ^^^^^^ scheme is dereferenced here, but checked against NULL later
73    fprintf(stderr, "%s:%d scheme of http resource is incorrect; "
74            "expected 'HTTP' but is %s\n",
75            __FILE__, __LINE__,
76            (scheme == NULL ? "NULL" : "invalid"));
77    res = CURLE_HTTP_RETURNED_ERROR;
78    goto test_cleanup;
79  }
```

I propose to either:
- add a test in the if-condition, to check against NULL to avoid the potential null-dereference
- remove the check against NULL in `(scheme == NULL ? "NULL" : "invalid")` if `scheme` can never be NULL

### CODE REVIEW

File: tests/libtest/lib1536.c [hunk 1]
```diff
@@ -69,7 +69,7 @@ int test(char *URL)
             __FILE__, __LINE__, res, curl_easy_strerror(res));
     goto test_cleanup;
   }
-  if(memcmp(scheme, "HTTP", 5) != 0) {
+  if(scheme == NULL || memcmp(scheme, "HTTP", 5) != 0) {
```
Comment (danielgustafsson) [inline tests/libtest/lib1536.c:72]: Agreed, makes sense.

**Review by danielgustafsson — APPROVED** [review]

**Comment by kokke** [comment]
Hi @danielgustafsson 

Thanks for super fast approval of both my PRs. 

I agree the problems "fixed" are benign. I would personally prefer a failed test-case with an error-message instead of a segfault, which is the main effect of this.

**Comment by kokke** [comment]
Hi @bagder in this case the segfault would be caused by the `memcmp` at line 72 (as mentioned in the comments) which, to my knowledge, dereferences its arguments.

**Review by emilengler — APPROVED** [review]
The C89 standard doesn't define anything for memcmp with a NULL pointer.
See: http://port70.net/~nsz/c/c89/c89-draft.html#4.11.4.1

**Comment by bagder** [comment]
I don't think `scheme` can be NULL at that point. How would that happen? And if it can, why hasn't that been detected by torture tests?

**Comment by jay** [comment]
> I don't think `scheme` can be NULL at that point. How would that happen? And if it can, why hasn't that been detected by torture tests?

According to [CURLINFO_SCHEME](https://curl.se/libcurl/c/CURLINFO_SCHEME.html) it can return NULL so technically he's correct however internally it may currently be an [impossibility](https://github.com/curl/curl/blob/curl-7_75_0/lib/connect.c#L611-L625) that it's NULL after the transfer.

**Comment by bagder** [comment]
@kokke sorry, I didn't mean to question your tool nor your fix, I was just trying to take a step back and think why the NULL check is there in the first place.

@jay: thanks for checking!

**Comment by kokke** [comment]
@bagder no offense taken at all. I understand if you guys see many questionable change proposals, so I just wanted to make my reasoning explicitly clear.

**Comment by bagder** [comment]
Thanks!

**Comment by kokke** [comment]
And thank you for kindly accepting all my small change proposals 👍 Contributing has been a very pleasant experience
